### ISSUE TITLE
newrelic_alert_condition type browser_metric uses the APM ID instead of Browser ID

### ISSUE THREAD

**Issue by mdenkinger** [newrelic/terraform-provider-newrelic#1016]
### Terraform Version
Terraform v0.12.26

### Affected Resource(s)
- `newrelic_alert_condition`
- type: `browser_metric`
- metrics: `end_user_apdex`, `page_views_with_js_errors`  

### Terraform Configuration
```hcl
provider "newrelic" {
  api_key       = local.new_relic_nerd_graph_api_key
  admin_api_key = local.new_relic_admin_api_key
  account_id    = local.new_relic_account_id
}
```

### Actual Behavior

A `newrelic_alert_condition` type `browser_metric` metic `end_user_apdex` changes to the APM entity ID instead of using the Browser entity ID. 

Output:
```hcl
  # module.some-newrelic-browser.newrelic_alert_condition.service_alert_browser_apdex_low will be updated in-place
  ~ resource "newrelic_alert_condition" "service_alert_browser_apdex_low" {
        condition_scope       = "application"
        enabled               = true
      ~ entities              = [
          - 874884861,
          + 910965338,
        ]
        id                    = "123456:987654321"
        metric                = "end_user_apdex"
        name                  = "some alert"
        policy_id             = 123456
        runbook_url           = "https://some.url"
        type                  = "browser_metric"
        violation_close_timer = 0

        term {
            duration      = 5
            operator      = "below"
            priority      = "critical"
            threshold     = 0.75
            time_function = "all"
        }
    }
```

Configuration
```hcl
resource "newrelic_alert_condition" "service_alert_browser_apdex_low" {
  policy_id = data.newrelic_alert_policy.some_ploicy.id

  name = "some ${local.service_name} browser apdex low"

  runbook_url = "https://some.url"

  type = "browser_metric"

  condition_scope = "application"

  entities = [
    data.newrelic_entity.browser.application_id,
  ]

  metric = "end_user_apdex"

    term {
    duration      = 5
    operator      = "below"
    priority      = "critical"
    threshold     = "0.75"
    time_function = "all"
  }
}

# 910965338
data "newrelic_entity" "browser" {
  name   = local.newrelic_browser_name
  domain = "BROWSER"
}

# 874884861
data "newrelic_entity" "apm" {
  name   = local.newrelic_browser_name
  domain = "APM" 
}
```

### Expected Behavior
A `newrelic_alert_condition` type `browser_metric` metric `end_user_apdex` uses the Browser entity ID as it is a `browser_metric`. 

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`
1. `terraform apply`

### Debug Output
_- none -_

### Panic Output
_- none -_

### Important Factoids
_- none -_

### References
_- none -_

**Comment by mdenkinger** [comment]
we could change the ID over to the associated APM Id, but that feels wrong, as it is a browser_metric, as well the New Relic selects the Browser entity and not the APM entity.

**Comment by zlesnr** [comment]
Hi, thank you for reporting the issue.  I'm not sure I follow what you are pointing out though.  The entities should be using the entity from the `data.newrelic_entity.browser.application_id` reference you have.  It looks like the right number is getting used based on the comment in your HCL there.

**Comment by mdenkinger** [comment]
Yes, the changes will be applied with the `data.newrelic_entity.browser.application_id` but within New Relic it is changed to the associated APM entity ID. 
Therefore this modification will be apply over and over again. You could fix this behaviour by using the APM entity Id, but as described, this looks wrong, as this is an alert for `browser_metric`s.

**Comment by zlesnr** [comment]
Okay, that seems strange.  I'll see if I can reproduce this today.

**Comment by zlesnr** [comment]
I'm able to reproduce the issue you are seeing.  At first thought, I'm thinking we dangle a `serving_apm_application_id` output so that you can reference the right thing and the code stops flapping.  Does that seem like a reasonable solution?  Or are is the request more that the platform not modify the ID?  If that is the case, then we'll want to run this up as a feature request in the platform.  Otherwise if we can just expose the other ID on the data source, perhaps that will get us past this.

**Comment by zlesnr** [comment]
I've put up a PR with the additional attribute you could reference to resolve the drift.

**Comment by mdenkinger** [comment]
>  Or are is the request more that the platform not modify the ID? If that is the case, then we'll want to run this up as a feature request in the platform. 
Yes, it's more that the platform not modify the ID. 

> Otherwise if we can just expose the other ID on the data source, perhaps that will get us past this.
Yes, we just used the serving APM application Id and it stopped flapping. But as I tried to point out, I believe this would wrong and should just work as workaround for now.

**Comment by zlesnr** [comment]
I'll see if I can figure out which team owns that and make the request.  Until then, the additional ID will be available in the next release.

**Comment by zlesnr** [comment]
The alerts team did some digging, and it sounds like things are operating as designed, and there are historical reasons for this, but I'm not clear on the details.  In any case, the new attribute we discussed has been released in `v2.13.4`.
